glue/viewers/image/composite_array.py

```diff
@@ -80,7 +101,24 @@ def __call__(self, bounds=None):
         img = None
         visible_layers = 0
 
-        for uuid in sorted(self.layers, key=lambda x: self.layers[x]['zorder']):
+        # Get a sorted list of UUIDs with the top layers last
+        sorted_uuids = sorted(self.layers, key=lambda x: self.layers[x]['zorder'])
+
+        # We first check that layers are either all colormaps or all single colors.
+        # In the case where we are dealing with colormaps, we can start from
+        # the last layer that has an opacity of 1 because layers below will not
+        # affect the output, assuming also that the colormaps do not change the
+        # alpha
+        if self.mode == 'colormap':
+            start = 0
+            for i in range(len(sorted_uuids)):
+                layer = self.layers[sorted_uuids[i]]
+                if layer['visible']:
+                    if layer['alpha'] * layer['cmap'](CMAP_SAMPLING)[:, 3].min() == 1:
+                        start = i
+            sorted_uuids = sorted_uuids[start:]
```

Comment:
Would it maybe be more efficient to loop over the layers backwards by inverse zorder?

Reply:
Good point, yes we can just stop at the highest opaque layer.